src/app/(proper_react)/redesign/(authenticated)/user/dashboard/page.tsx

```diff
@@ -34,9 +35,24 @@ export default async function DashboardPage() {
 
   const result = await getOnerepProfileId(session.user.subscriber.id);
   const profileId = result[0]["onerep_profile_id"] as number;
+  const brokerScanReleaseDateParts = (
+    process.env.BROKER_SCAN_RELEASE_DATE ?? ""
+  ).split("-");
+  if (brokerScanReleaseDateParts[0] === "") {
+    brokerScanReleaseDateParts[0] = "2023";
+  }
+  const brokerScanReleaseDate = new Date(
+    Date.UTC(
+      Number.parseInt(brokerScanReleaseDateParts[0], 10),
+      Number.parseInt(brokerScanReleaseDateParts[1] ?? "12", 10) - 1,
+      Number.parseInt(brokerScanReleaseDateParts[2] ?? "05", 10),
+    ),
+  );
   if (
     !profileId &&
-    canSubscribeToPremium({ user: session?.user, countryCode: countryCode })
+    canSubscribeToPremium({ user: session?.user, countryCode: countryCode }) &&
+    (parseIso8601Datetime(session.user.subscriber.created_at)?.getTime() ?? 0) >
+      brokerScanReleaseDate.getTime()
```

Comment:
We would also need to check if a user already `hasPremium` and if they did not run a scan we are supposed to redirect as well.

Reply:
Also, might want to move that check to the dashboard’s `layout.tsx` and remove the one in https://github.com/mozilla/blurts-server/blob/MNTOR-2264-welcome-new-user/src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/layout.tsx#L24.

Reply:
Added in 75f0d4fae93e35cf3f951d2acdbcb851cd098cfc.

As for the layout, I ended up removing that completely in 0d8f9c45c08698b5c3795f936b6db6f821e5adb1 - I don't think the redirect adds that much value there, since users shouldn't even end up in that flow in that case, and other than that the layout didn't do anything that a parent layout didn't already do. Let me know if I'm missing some obvious reason why I shouldn't have done that :)

Reply:
Thanks! I don’t think it does anything we would not already have handled in higher routes.